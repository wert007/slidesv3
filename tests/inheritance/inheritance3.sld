abstract struct Animal {
    v: int;
}

struct Cat: Animal {
    func $constructor() {
        this.v = 0;
    }

    func a() {
        this.v = 999;
    }

    func $toString() -> string {
        return base + 'Im a Cat!';
    }
}


func main() {
    let a: Animal = new Cat();
    print(a.v == 0);
    let ac = cast a: Cat;
    if ac {
        ac.a();
        print(ac);
    }
    print(a.v == 999);
}