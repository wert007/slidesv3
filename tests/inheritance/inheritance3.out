true
struct Animal {
    v = 999,
}Im a Cat!
true
= 0
